### PR TITLE
ci: Add environment files for all settings

### PR DESCRIPTION
**Pull request by maflcko**
This moves all environment settings from travis to files in the ci folder. Now, it is possible to easily run each travis configuration with a single command.

### CODE REVIEW

**Review by promag — COMMENTED** [review]
This approach pollutes the calling bash environment. Should work without `export`?

Concept ACK.

File: .travis.yml [hunk 1]
```diff
@@ -91,101 +91,49 @@ jobs:
...
       name: 'macOS 10.10  [GOAL: deploy] [no functional tests]'
       env: >-
-        HOST=x86_64-apple-darwin14
-        PACKAGES="cmake imagemagick libcap-dev librsvg2-bin libz-dev libbz2-dev libtiff-tools python3-dev python3-setuptools"
-        OSX_SDK=10.11
-        RUN_UNIT_TESTS=false
-        RUN_FUNCTIONAL_TESTS=false
-        GOAL="deploy"
-        BITCOIN_CONFIG="--enable-gui --enable-reduce-exports --enable-werror"
+        FILE_ENV="./ci/test/00_setup_env_mac.sh"
```
Comment (fanquake) [inline .travis.yml]: Should this be `00_setup_env_amd64_mac` ? There's no `00_setup_env_mac.sh` file.

**Review by fanquake — COMMENTED** [review]
Concept ACK

I'll run through each of the different ENV builds locally. Have done `amd64_asan` and `amd64_qt5` so far.

For a future change / discussion, it'd be handy if the Docker containers were named. `amd64_qt` is more useful than `peaceful_rubin` once there are a bunch floating around.

Also, maybe we could stop (although not destroy) containers automatically after successful completion of the tests?

**Comment by MarcoFalke** [comment]
> This approach pollutes the calling bash environment.

Can you explain what command exactly does that? I can't find any.

It will potentially pollute your ccache dir and the git repo with build artifacts, but that has nothing to do with bash.

File: .travis.yml (hunk 1 above)
Comment (MarcoFalke) [reply .travis.yml]: Fixed

**Comment by MarcoFalke** [comment]
Good suggestions @fanquake. I'll probably leave those as follow-up features that can be added.

**Comment by Sjors** [comment]
ACK. I would find it easier if the new files were numbered the same as its equivalent Travis machine.

Recipe to verify the move:

1. delete `export ` from `ci/test/00_setup_env_*`
2.`git commit -a --amend`
3.  `git show --color-moved=dimmed-zebra --color-moved-ws=allow-indentation-change`

I tried (cloned in /home/travis/bitcoin):
```sh
cd bitcoin
FILE_ENV="./ci/test/00_setup_env_amd64_qt5.sh" ./ci/test_run_all.sh
```

This fails at the configure step:
```
travis_fold:start:configure
configure: loading site script /home/travis/bitcoin/depends/x86_64-unknown-linux-gnu/share/config.site
configure: loading cache config.cache
configure: error: `CXXFLAGS' was not set in the previous run
configure: error: `CFLAGS' was not set in the previous run
configure: error: in `/home/travis/bitcoin/build':
configure: error: changes in the environment can compromise the build
configure: error: run `make distclean' and/or `rm config.cache' and start over
...
configure:2987: loading cache config.cache
configure:3014: error: `CXXFLAGS' was not set in the previous run
configure:3014: error: `CFLAGS' was not set in the previous run
configure:3051: error: in `/home/travis/bitcoin/build':
configure:3053: error: changes in the environment can compromise the build
configure:3055: error: run `make distclean' and/or `rm config.cache' and start over
...
configure: exit 1
```

It may have gotten poluted from a previous run, so I tried purging docker instances and nuking the git repository:

```
docker stop $(docker ps -a -q)
docker container prune
git clean -dfx
```

Now it runs.

As a followup, it would be nice if you can run all configurations with a single command (stopping if any fails).

**Comment by MarcoFalke** [comment]
> Now it runs. Because Docker has root privileges, things very quickly turn into permission hell if you try to run more than one instance.

You can't run more than one ci instance on the repo at the same time.

> As a followup, it would be nice if you can run all configurations with a single command (stopping if any fails).

I'll leave this as follow-up ;)

**Comment by Sjors** [comment]
> You can't run more than one ci instance on the repo at the same time.

I was running them in sequence.

**Comment by MarcoFalke** [comment]
That should work as long as you run `sudo git clean -dffx` (at your own risk, obviously)

**Pull request merged**